### PR TITLE
CPM-549: Migration fixes

### PR DESCRIPTION
**Pull request by mmetayer**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Bundle/Command/MigrateToUuid/MigrateToUuidCreateColumns.php [hunk 1]
```diff
@@ -96,5 +96,17 @@ private function addUuidColumnAndIndexOnUuid(string $tableName, string $uuidColu
         );
 
         $this->connection->executeQuery($addUuidColumnAndIndexOnUuidQuery);
+
+        if ('pim_versioning_version' === $tableName) {
+            // TODO: remove this index once the pim_versioning_version is fully migrated
+            $this->connection->executeQuery(
+                <<<SQL
+                ALTER TABLE `pim_versioning_version`
+                ADD INDEX `resource_name_uuid_id_idx` (`resource_name`, `resource_uuid`, `resource_id`),
```
Comment (pierallard) [inline src/Akeneo/Pim/Enrichment/Bundle/Command/MigrateToUuid/MigrateToUuidCreateColumns.php:105]: Can you call it `resource_name_resource_uuid_version_idx`
I tried to be consistent in the naming of all the indexes I added here
https://github.com/akeneo/pim-community-dev/pull/16643/files#diff-82a45190efc33f7adad182ec0f1e4e32a821fecc4fff71a8a18f6909cb202d41R210

I added myself a comment to drop mine once you merged yours.

File: src/Akeneo/Pim/Enrichment/Bundle/Command/MigrateToUuid/MigrateToUuidCreateColumns.php [hunk 2]
```diff
@@ -96,5 +96,17 @@ private function addUuidColumnAndIndexOnUuid(string $tableName, string $uuidColu
         );
 
         $this->connection->executeQuery($addUuidColumnAndIndexOnUuidQuery);
+
+        if ('pim_versioning_version' === $tableName) {
+            // TODO: remove this index once the pim_versioning_version is fully migrated
```
Comment (pierallard) [inline src/Akeneo/Pim/Enrichment/Bundle/Command/MigrateToUuid/MigrateToUuidCreateColumns.php:101]: In fact, there is a (`resource_name`, `resource_id`, `resource_id`) index, so don't need to drop this index once migration is done, we will keep it definitely.

**Review by nmarniesse — APPROVED** [review]

**Review by pierallard — APPROVED** [review]

File: src/Akeneo/Pim/Enrichment/Bundle/Command/MigrateToUuid/MigrateToUuidCreateColumns.php (hunk 1 above)
Comment (mmetayer) [reply src/Akeneo/Pim/Enrichment/Bundle/Command/MigrateToUuid/MigrateToUuidCreateColumns.php:105]: nope, it's not the same index (does not index the version column), but I can call it resource_name_resource_uuid_resource_id_idx if you want

File: src/Akeneo/Pim/Enrichment/Bundle/Command/MigrateToUuid/MigrateToUuidCreateColumns.php [hunk 3]
```diff
@@ -96,5 +96,17 @@ private function addUuidColumnAndIndexOnUuid(string $tableName, string $uuidColu
...
+            $this->connection->executeQuery(
+                <<<SQL
+                ALTER TABLE `pim_versioning_version`
```
Comment (pierallard) [inline src/Akeneo/Pim/Enrichment/Bundle/Command/MigrateToUuid/MigrateToUuidCreateColumns.php:104]: migrate_to_uuid_temp_index_to_delete

**Review by pierallard — COMMENTED** [review]
zefzef

**Pull request merged**